src/Uno.UI/UI/Xaml/Controls/TreeView/TreeViewItem.cs

```diff
@@ -0,0 +1,776 @@
+﻿using System;
+using Windows.ApplicationModel.DataTransfer;
+using Windows.System;
+using Windows.UI.Core;
+using Windows.UI.Xaml.Automation;
+using Windows.UI.Xaml.Automation.Peers;
+using Windows.UI.Xaml.Controls.Primitives;
+using Windows.UI.Xaml.Input;
+using Windows.UI.Xaml.Media;
+using TreeNodeSelectionState = Windows.UI.Xaml.Controls.TreeViewNode.TreeNodeSelectionState;
+
+namespace Windows.UI.Xaml.Controls
+{
+	public partial class TreeViewItem : ListViewItem
+	{
+		private const long c_dragOverInterval = 1000 * 10000;
+		private const string c_multiSelectCheckBoxName = "MultiSelectCheckBox";
+		private const string c_expandCollapseChevronName = "ExpandCollapseChevron";
+		
+		private bool m_expansionCycled;
+		private CheckBox m_selectionBox;
+		private DispatcherTimer m_expandContentTimer;
+		private TreeView m_ancestorTreeView;
+		private UIElement m_expandCollapseChevron;
+
+		public TreeViewItem()
+		{
+			DefaultStyleKey = typeof(TreeViewItem);
+			SetValue(TreeViewItemTemplateSettingsProperty, new TreeViewItemTemplateSettings());
+		}
+
+		protected override void OnKeyDown(KeyRoutedEventArgs e)
+		{
+			var targetNode = TreeNode;
+			if (targetNode != null)
+			{
+				var treeView = AncestorTreeView;
+				var key = e.Key;
+				var originalKey = e.OriginalKey;
+
+				if (originalKey == VirtualKey.GamepadA &&
+					treeView.ListControl.IsMultiselect)
+				{
+					HandleGamepadAInMultiselectMode(targetNode);
+					e.Handled = true;
+				}
+				else if (IsInReorderMode(key))
+				{
+					HandleReorder(key);
+					e.Handled = true;
+				}
+				// Collapse/Expand
+				else if (IsExpandCollapse(key))
+				{
+					var handled = HandleExpandCollapse(key);
+					if (handled)
+					{
+						e.Handled = true;
+					}
+				}
+				else if (key == VirtualKey.Space &&
+						 treeView.ListControl.IsMultiselect)
+				{
+					var selectionBox = m_selectionBox;
+					bool isSelected = CheckBoxSelectionState(selectionBox) == TreeNodeSelectionState.Selected;
+					selectionBox.IsChecked = !isSelected;
+					e.Handled = true;
+				}
+			}
+
+			// Call to base for all other key presses
+			base.OnKeyDown(e);
+		}
+
+		protected override void OnDrop(DragEventArgs args)
+		{
+			if (!args.Handled && args.AcceptedOperation == DataPackageOperation.Move)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Refactor this method to add validation of parameter 'args' before using it.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=5156819)